src/field/field3d.cxx

```diff
@@ -150,54 +135,53 @@ Field3D* Field3D::timeDeriv() {
 void Field3D::splitYupYdown() {
   TRACE("Field3D::splitYupYdown");
   
-  if((yup_field != this) && (yup_field != nullptr))
+  if (!yup_fields.empty()) {
     return;
+  }
 
-  // yup_field and ydown_field null
-  yup_field = new Field3D(fieldmesh);
-  ydown_field = new Field3D(fieldmesh);
+  for (int i = 0; i < fieldmesh->ystart; ++i) {
+    yup_fields.emplace_back(fieldmesh);
```

Comment:
We're passing a mesh here to the `emplace_back` method of a `std::vector<Field3D>` -- what does this do? Does it work out how to call one of the `Field3D` constructors, what if we want to specify other parameters (e.g. location in the future)?

Reply:
Yes, `emplace_back` takes `class... Args` and `forward`s them to the member type ctor:
https://en.cppreference.com/w/cpp/container/vector/emplace_back

Should be trivial to add other arguments